plugins/modules/ec2_eip.py

```diff
@@ -76,6 +76,24 @@
       - Allocates the new Elastic IP from the provided public IPv4 pool (BYOIP)
         only applies to newly allocated Elastic IPs, isn't validated when O(reuse_existing_ip_allowed=true).
     type: str
+  update_reverse_dns_record:
+    description: Whether to update the reverse DNS record of ec2 Elastic IP Address (eip)
+    required: false
+    type: bool
+  allocation_id:
+    description: The allocation ID of ec2 EIP.
+    required: false
+    type: str
+  domain_name:
+    description: The domain name to modify for the IP address.
+    required: false
+    type: str
```

Comment:
```suggestion
  domain_name:
    description: The domain name to modify for the IP address.
    required: false
    type: str
```

only the `domain_name` should be required. If you create a new EIP the API will return the allocation ID, in case you are trying to modify existing IP the Pulic IP must be provided to retrieve te allocation ID.

Reply:
updated the code to work with domain_name and public_ip only.
Not sure how to achieve result without using `update_reverse_dns_record`, do we want to combine the functionality of [modify_address_attribute](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/modify_address_attribute.html) into same task as creating IP or keep it as independent task?

```
- name: Create new EIP and update DNS Record
  ec2_eip:
     state: present
     domain_name: example.net
     public_ip: 1.2.3.4
```
or
```
- name: Update PTR Record of existing EIP
  ec2_eip:
     update_reverse_dns_record: true
     public_ip: 1.2.3.4
     domain_name: example.net
```

Reply:
`update_reverse_dns_record` is not required.
The module should work like any other module.
Currently if you specified the `public_ip` it means that you are trying to modify existing allocated EIP and if the `domain_name` is specified along with the module should also update this attribute
When the IP is not specified a new one will be created and the domain_name linked to this new EIP

Reply:
updated the code.